testpar/t_filters_parallel.c

```diff
@@ -385,8 +458,388 @@ verify_space_alloc_status(hid_t dset_id, hid_t dcpl_id, num_chunks_written_t chu
                 MPI_Abort(MPI_COMM_WORLD, 1);
         }
     }
+}
+
```

Comment:
The vast majority of changes in this file were just adapting the code to use new functions for creating/open/reading/writing multiple datasets at once rather than a single dataset. The new `verify_chunk_opt_status` function below tries to verify that the library is returning the correct value for whether linked-chunk, multi-chunk or non-optimized I/O was performed.